### ISSUE TITLE
New rule: No unnecessary regex global flags

### ISSUE THREAD

**Issue by tom-sherman** [eslint/eslint#10879]
I recently came across an ugly pattern in Regex regarding the `g` flag. See here for details: https://stackoverflow.com/questions/2630418/javascript-regex-returning-true-then-false-then-true-etc

Would have been nice for the linter to warn that I'm using a global flag when I was using `^` and `$` characters at the beginning and end of the regex.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[x] Warns about a potential error
[x] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
/^test$/g // Warn
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It could prevent quite a common issue caused by weirdness in the Regex global flag.

Originally raised in https://github.com/standard/standard/issues/1152
